### PR TITLE
Viewset migration

### PR DESCRIPTION
**Pull request by whitej6**

### CODE REVIEW

File: nautobot_bgp_models/forms.py [hunk 1]
```diff
@@ -163,9 +163,15 @@ class Meta:
         ]
 
 
-class PeeringRoleForm(
```
Comment (mzbroch) [inline nautobot_bgp_models/forms.py:166]: we probably need this one for viewset migration

File: nautobot_bgp_models/models.py [hunk 2]
```diff
@@ -318,7 +325,8 @@ class PeerGroup(PrimaryModel, InheritanceMixin, BGPExtraAttributesMixin):
 
     name = models.CharField(max_length=100)
 
-    template = models.ForeignKey(
+    # Rename to avoid clash with DRF renderer
+    peergroup_template = models.ForeignKey(
```
Comment (mzbroch) [inline nautobot_bgp_models/models.py:329]: following code has to be udpated in `forms.py` 

```
            for t in self.cleaned_data.get("template", []):  # pylint: disable=invalid-name
                models.PeerGroup.objects.create(
                    name=t.name,
                    template=t,
                    routing_instance=obj,
                )
```
based on selected master template, it auto-populates locally significant peer groups

File: nautobot_bgp_models/urls.py [hunk 3]
```diff
@@ -1,177 +1,44 @@
...
-        kwargs={"model": models.PeerEndpoint},
-    ),
     path(
         "peer-endpoints/<uuid:pk>/extra-attributes/",
         views.BgpExtraAttributesView.as_view(),
         name="peerendpoint_extraattributes",
         kwargs={"model": models.PeerEndpoint},
     ),
-    path("peerings/", views.PeeringListView.as_view(), name="peering_list"),
-    path("peerings/add/", views.PeeringAddView.as_view(), name="peering_add"),
```
Comment (mzbroch) [inline nautobot_bgp_models/urls.py:148]: This view no longer works as of now

File: nautobot_bgp_models/__init__.py [hunk 4]
```diff
@@ -23,7 +23,7 @@ class NautobotBGPModelsConfig(PluginConfig):
     description = "Nautobot BGP Models Plugin."
     base_url = "bgp"
     required_settings = []
-    min_version = "1.3.0"
+    min_version = "1.5.0"
```
[the remark this answers is not in the document]
Comment (whitej6) [reply nautobot_bgp_models/__init__.py]: So this is a discussion we need to have internally on constrains vs plugin config. This is different than CI file.

File: nautobot_bgp_models/urls.py (hunk 3 above)
Comment (whitej6) [reply nautobot_bgp_models/urls.py:148]: Yeah... Need to revisit this part.

File: docs/index.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ The plugin is available as a Python package in PyPI and can be installed with `p
 pip install nautobot-bgp-models
 ```
 
-> The plugin is compatible with Nautobot 1.3 and higher
+> The plugin is compatible with Nautobot 1.4 and higher
```
Comment (whitej6) [inline docs/index.md:17]:
```suggestion
> The plugin is compatible with Nautobot 1.5.4 and higher
```

File: development/nautobot_config.py [hunk 6]
```diff
@@ -83,239 +112,33 @@ def is_truthy(arg):
...
+#
 
 # Enable installed plugins. Add the name of each plugin to the list.
 PLUGINS = ["nautobot_bgp_models"]
 
 # Plugins configuration settings. These settings are used by various plugins that the user may have installed.
 # Each key in the dictionary is the name of an installed plugin and its value is a dictionary of settings.
 # PLUGINS_CONFIG = {
-#     'my_plugin': {
+#     '{{cookiecutter.plugin_name}}': {
```
Comment (whitej6) [inline development/nautobot_config.py:140]:
```suggestion
#     'nautobot_bgp_models': {
```

**Comment by mzbroch** [comment]
Depends on #90

**Comment by mzbroch** [comment]
@whitej6 I am continuing this work in the https://github.com/nautobot/nautobot-plugin-bgp-models/pull/96

**Comment by mzbroch** [comment]
Replaced by #96

**Pull request closed without merging**